pkg/controller/deliverable/reconciler.go

```diff
@@ -101,7 +121,9 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 			r.conditionManager.AddPositive(TemplateStampFailureCondition(typedErr))
 		case realizer.ApplyStampedObjectError:
 			r.conditionManager.AddPositive(TemplateRejectedByAPIServerCondition(typedErr))
-			err = controller.NewUnhandledError(err)
+			if !kerrors.IsForbidden(typedErr.Err) {
```

Comment:
Is there a way to avoid bringing kerrors import back to reconciler? The answer can be no. But we were able to get rid of it by pushing IsNotFound to repository.

Reply:
I think we can if we do this check in component.go [here](https://github.com/vmware-tanzu/cartographer/blob/2200c58d172a7bd3ca17e759fb5ac949b0cd4b0f/pkg/realizer/deliverable/component.go#L121) and return a different error type. Then we'd just have to add a separate case to the reconciler.

If we want to keep kerrors out of the reconciler it shouldn't be too hard to make that change.